### ISSUE TITLE
consumer is incompatible with grpc-protobuf version 1.59.0

### ISSUE THREAD

**Issue by austin226** [pact-foundation/pact-jvm#1816]
When using:
* pact-consumer 4.6.14
* pact-driver core 0.4.2
* grpc 1.59.0
* `au.com.dius.pact.consumer.model.MockProviderConfig` with `transportRegistryEntry = "protobuf/transport/grpc"`

the following error occurs:

```
java.lang.NoClassDefFoundError: io/grpc/internal/AbstractManagedChannelImplBuilder
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
	at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(Unknown Source)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Unknown Source)
	at io.grpc.ManagedChannelRegistry.getHardCodedClasses(ManagedChannelRegistry.java:142)
	at io.grpc.ManagedChannelRegistry.getDefaultRegistry(ManagedChannelRegistry.java:103)
	at io.grpc.ManagedChannelProvider.provider(ManagedChannelProvider.java:43)
	at io.grpc.ManagedChannelBuilder.forTarget(ManagedChannelBuilder.java:86)
	at io.pact.plugins.jvm.core.DefaultPluginManager.tryInitPlugin(PluginManager.kt:884)
	at io.pact.plugins.jvm.core.DefaultPluginManager.initialisePlugin(PluginManager.kt:866)
	at io.pact.plugins.jvm.core.DefaultPluginManager.loadPlugin(PluginManager.kt:419)
	at au.com.dius.pact.consumer.dsl.PactBuilder.usingPlugin(PactBuilder.kt:113)
``` 

Additionally,
```
Did not find an entry for 'protobuf/transport/grpc' in the plugin registry
au.com.dius.pact.consumer.InvalidMockServerRegistryEntry: Did not find an entry for 'protobuf/transport/grpc' in the plugin registry
	at app//au.com.dius.pact.consumer.PluginMockServer.start(PluginMockServer.kt:41)
	at app//au.com.dius.pact.consumer.BaseMockServer.runAndWritePact(MockHttpServer.kt:141)
```

**Comment by austin226** [comment]
May depend on core changes - https://github.com/pact-foundation/pact-plugins/issues/72

**Comment by rholshausen** [comment]
Looks like that refactor upgraded gRPC libs from 1.56.1 to 1.64.0, and 1.59.0 has a breaking change notice https://github.com/grpc/grpc-java/releases/tag/v1.59.0

You could try add an exclusion on pact-driver core 0.4.2 and then add pact-driver core 0.4.1 as a dependency

**Comment by rholshausen** [comment]
Oh, wait, that refactor has not been released yet. Would that fix your issue (upgrading gRPC libs to 1.64)?

**Comment by austin226** [comment]
I see [pact driver 0.5.0](https://github.com/pact-foundation/pact-plugins/releases/tag/pact-jvm-driver-v0.5.0) was released - matching grpc libs to 1.66 here might fix the issue

**Comment by rholshausen** [comment]
[4.6.14](https://github.com/pact-foundation/pact-jvm/releases/tag/4_6_14) and [4.5.13](https://github.com/pact-foundation/pact-jvm/releases/tag/4_5_13) have been released which use pact driver 0.5.1

**Comment by austin226** [comment]
Upgrading to 4.6.14 solved my issue - thanks!
